packages/plans-grid/src/plans-table/index.tsx

```diff
@@ -43,15 +44,19 @@ const PlansTable: React.FunctionComponent< Props > = ( {
 	currentDomain,
 	disabledPlans,
 	locale,
+	billingPeriod,
+	onMaxMonhtlyDiscountPercentageChange,
 	showTaglines = false,
 	CTAVariation = 'NORMAL',
 	popularBadgeVariation = 'ON_TOP',
 	customTagLines,
 	defaultAllPlansExpanded = false,
 } ) => {
-	const supportedPlans = useSelect( ( select ) =>
-		select( PLANS_STORE ).getSupportedPlans( locale )
-	);
+	const { supportedPlans, maxAnnualDiscount } = useSupportedPlans( locale, billingPeriod );
+
+	React.useEffect( () => {
+		onMaxMonhtlyDiscountPercentageChange( maxAnnualDiscount );
+	}, [ onMaxMonhtlyDiscountPercentageChange, maxAnnualDiscount ] );
```

Comment:
If I understand correctly, if you use the hook as I commented above, we won't need this.